userspace/libsinsp/filterchecks.h

```diff
@@ -118,14 +118,20 @@ class sinsp_filter_check : public gen_event_filter_check
 	// Extract the field from the event. In sanitize_strings is true, any
 	// string values are sanitized to remove nonprintable characters.
 	//
-	uint8_t* extract(gen_event *evt, OUT uint32_t* len, bool sanitize_strings = true);
-	virtual uint8_t* extract(sinsp_evt *evt, OUT uint32_t* len, bool sanitize_strings = true) = 0;
+	bool extract(gen_event *evt, OUT vector<extract_value_t>& values, bool sanitize_strings = true);
+
+	// Alias of extract that uses the sinsp_evt type.
+	// By default, this fills the vector with only one value, retireved by calling the single-result
+	// extract method.
+	// If a NULL value is returned by extract, the vector is emptied.
+	// Subclasses are meant to either override this, or the single-valued extract method.
+	virtual bool extract(sinsp_evt *evt, OUT vector<extract_value_t>& values, bool sanitize_strings = true);
 
 	//
 	// Wrapper for extract() that implements caching to speed up multiple extractions of the same value,
 	// which are common in Falco.
 	//
-	uint8_t* extract_cached(sinsp_evt *evt, OUT uint32_t* len, bool sanitize_strings = true);
+	bool extract_cached(sinsp_evt *evt, OUT vector<extract_value_t>& values, bool sanitize_strings = true);
```

Comment:
What about using only extract with `bool cached = false` default argument?
I think it would feel better.

Reply:
The problem here is that `extract_cached` has a default implementation in the `sinsp_filter_check`, whereas the `extract` method is overridden by all its sub-classes. That way, the extraction logic stays only in one place, which I kind of like. Plus, the optional boolean value would be alongside `sanitize_strings`, and having two optional booleans in the same method can still become confusing IMO.

Reply:
Makes sense; thanks!